From a59bef58d9ddc54ad69f0ee802666804ae2fbc7b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Fri, 23 Aug 2024 18:42:43 +0200
Subject: disable PIE, remove BRANCH_VERSION

 * PIE is unsupported on Haiku, but not detected correctly.
 * the BRANCH_VERSION detection is faulty because it checks whether
   a .git directory exists, which is the case in haikuporter.
   We build from a tarball, so remove that code entirely.

diff --git a/configure.ac b/configure.ac
index c43ef39..65f8ce5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -45,7 +45,6 @@ dnl until flex gets their act together, use pedantic instead of pedantic-errors
 AX_CHECK_COMPILE_FLAG([-pedantic], [CFLAGS+=" -pedantic"])
 dnl AX_CHECK_COMPILE_FLAG([-pedantic-errors], [CFLAGS+=" -pedantic-errors"])
 AX_CHECK_COMPILE_FLAG([-fstack-protector-strong], [CFLAGS+=" -fstack-protector-strong"])
-AX_CHECK_COMPILE_FLAG([-fPIE], [CFLAGS+=" -fPIE"])
 AX_CHECK_COMPILE_FLAG([-Wformat=2], [CFLAGS+=" -Wformat=2"])
 AX_CHECK_COMPILE_FLAG([-Werror=format-security], [CFLAGS+=" -Werror=format-security"])
 AX_CHECK_COMPILE_FLAG([-Wstrict-prototypes], [CFLAGS+=" -Wstrict-prototypes"])
@@ -55,10 +54,7 @@ AX_CHECK_COMPILE_FLAG([-Wextra], [CFLAGS+=" -Wextra"])
 AX_CHECK_COMPILE_FLAG([-Wno-gnu-statement-expression], [CFLAGS+=" -Wno-gnu-statement-expression"])
 AX_CHECK_LINK_FLAG([-Wl,-z,relro], [LDFLAGS+=" -Wl,-z,relro"])
 AX_CHECK_LINK_FLAG([-Wl,-z,now], [LDFLAGS+=" -Wl,-z,now"])
-AX_CHECK_LINK_FLAG([-fPIE], [LDFLAGS+=" -fPIE"])
-AX_CHECK_LINK_FLAG([-pie], [LDFLAGS+=" -pie"])
 
-AC_HEADER_STDC
 
 gl_FUNC_REALPATH_WORKS
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 5059209..520352e 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -27,25 +27,6 @@ gregorio__GREGORIO_EXE_SUFFIX__SOURCES = \
 	gregoriotex/gregoriotex-write.c gregoriotex/gregoriotex-position.c \
 	gregoriotex/gregoriotex.h
 
-@MK@ifneq ($(wildcard ../.git),)
-@MK@  _tag_ = $(shell git describe --exact-match HEAD 2>/dev/null)
-@MK@  ifeq ($(_tag_),)
-@MK@    DEFS += -DBRANCH_VERSION='"$(shell git rev-parse --abbrev-ref HEAD)-$(shell git rev-parse --short HEAD)-$(shell git rev-list HEAD --count)"'
-@MK@  endif
-@MK@else
-@MK@  ifneq ($(wildcard ../.hg/git),)
-@MK@    _gitnode_ = $(shell hg log --template='{gitnode}' -r .)
-@MK@    ifeq ($(_gitnode_),)
-@MK@      DEFS += -DBRANCH_VERSION='"$(shell hg log --template='{activebookmark}-hg:{manifest}' -r .)"'
-@MK@    else
-@MK@      _tag_ = $(shell git --git-dir=../.hg/git describe --exact-match '$(_gitnode_)' 2>/dev/null)
-@MK@      ifeq ($(_tag_),)
-@MK@        DEFS += -DBRANCH_VERSION='"$(shell hg log --template='{activebookmark}' -r .)-$(shell git --git-dir=../.hg/git rev-parse --short '$(_gitnode_)')-$(shell git --git-dir=../.hg/git rev-list '$(_gitnode_)' --count)"'
-@MK@      endif
-@MK@    endif
-@MK@  endif
-@MK@endif
-
 # gabc files
 gregorio__GREGORIO_EXE_SUFFIX__SOURCES += \
 	gabc/gabc-elements-determination.c gabc/gabc-write.c \
-- 
2.45.2

